apps/src/codebridge/FileBrowser/prompts/openImportFromBackpackPrompt.tsx

```diff
@@ -0,0 +1,160 @@
+  newFile: NewFileFunction;
+  saveFile: SaveFileFunction;
+  projectFiles: MultiFileSource['files'];
+  validationFile?: ProjectFile;
+};
+
+export const openImportFromBackpackPrompt = async ({
+  dialogControl,
+  backpackApi,
+  newFile,
+  saveFile,
+  projectFiles,
+  validationFile,
+}: OpenImportFromBackpackPromptArgsType) => {
+  const handleError = (message: string) => () => {
+    // TODO: send analytics / add logging.
+    console.error(message);
+  };
+  // Delete a file from the backpack.
+  const handleDelete = async (filename: string) => {
+    backpackApi.deleteFiles(
+      [filename],
+      handleError(`Error in deleting file ${filename}`),
+      () => console.log(`Deleted file ${filename}`)
+    );
+  };
+
+  // Fetch file content from backpack and then update or create a project file.
+  const fetchFileContentAndProcess = (
+    fileName: string,
+    newFileName?: string
+  ) => {
+    backpackApi.fetchFile(
+      fileName,
+      handleError(`Error in fetching file ${fileName}`),
+      (fileContent: string) => {
+        if (newFileName) {
+          newFile({fileName: newFileName, contents: fileContent});
+        } else {
+          const fileId = Object.keys(projectFiles).find(
+            id => projectFiles[id].name === fileName
+          );
+          if (fileId) saveFile(fileId, fileContent);
+        }
+      }
+    );
+  };
+
+  // TODO: Adding a waiting UI (spinner) while waiting for backpack list retrieval.
+  backpackApi.getFileList(
+    handleError('Error in getting backpack file list.'),
+    async (filenames: string[]) => {
+      if (filenames.length === 0) {
+        dialogControl?.showDialog({
+          type: DialogType.GenericAlert,
+          title: 'Files Saved in Backpack',
+          message: 'Files saved to your backpack will appear here.',
+        });
+      } else {
+        const savedFilesInBackpack: GenericDropdownProps['items'] =
+          filenames.map(filename => ({value: filename, text: filename}));
+
+        const results = await dialogControl?.showDialog({
+          type: DialogType.GenericDropdown,
+          title: 'Files Saved in Backpack',
+          dropdownLabel: '',
+          confirmText: 'Import to project',
+          items: savedFilesInBackpack,
+          selectedValue: savedFilesInBackpack[0].value,
+          neutralText: 'Delete file from backpack',
+        });
+
+        if (results.type === 'cancel') return;
+
+        const selectedBackpackFileName = extractUserInput(results, true);
+        // Import backpack file to project.
+        if (results.type === 'confirm') {
+          let newFileName = selectedBackpackFileName;
+          let validateError = validateFileName({
+            fileName: newFileName,
+            folderId: DEFAULT_FOLDER_ID,
+            projectFiles,
+            isStartMode: false,
+            validationFile,
+          });
+          const isSupportFileName = validateError?.includes('support code');
```

Comment:
This won't work if the user is a non-english language. We may want to do validation a little differently so we don't have to do string parsing. Another potential issue is you can get a validation error for an invalid file name (in theory there should be no invalid file names in the backpack), which would then cause an infinite loop below because we would keep adding a number to a filename that continues to be invalid. 

So instead of reusing the method, we may instead want to just use `isDuplicateFilename`, which returns an enum indicating whether it is support or not.

Reply:
Great point! Makes sense to use `isDuplicateFilename` and will update in this PR.